userspace/falco/app_actions/init_inspectors.cpp

```diff
@@ -48,12 +48,11 @@ static void init_syscall_inspector(
 
 	if(!opts.all_events)
 	{
-		// Drop EF_DROP_SIMPLE_CONS kernel side
-		inspector->set_simple_consumer();
-		// Eventually, drop any EF_DROP_SIMPLE_CONS event
-		// that reached userspace (there are some events that are not syscall-based
-		// like signaldeliver, that have the EF_DROP_SIMPLE_CONS flag)
-		inspector->set_drop_event_flags(EF_DROP_SIMPLE_CONS);
+		m_state->ppm_sc_of_interest = inspector->enforce_simple_ppm_sc_set();
+		m_state->tp_of_interest = inspector->enforce_sinsp_state_tracepoints();
+		// We are not interested in sched_switch tracepoint,
+		// highly noisy and not useful for state/events enrichment.
+		m_state->tp_of_interest.erase(SCHED_SWITCH);
```

Comment:
Maybe we can remove it directly from the `sinsp_state_set` since it is not so necessary and quite noisy for every client :thinking:

Reply:
Yep i think that could be a neat idea. It seems like sched_switch events count between 5-10% of the total number of events, so we could save both "wake up time" and ring buffer space for everyone.

Reply:
Totally agree on this

Reply:
I think we should let clients remove it, just like Falco is doing, because `sched_switch` somewhat enriches sinsp state, but they enrich something that is unused by Falco.
And of course dropping it is a good perf tradeoff.